packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmissionStepper.tsx

```diff
@@ -0,0 +1,243 @@
+    title: 'Select Evaluation Queue',
+    confirmStep: StepsEnum.SUBMISSION_SUCCESS,
+    confirmButtonText: 'Submit',
+  },
+  SUBMISSION_SUCCESS: {
+    id: StepsEnum.SUBMISSION_SUCCESS,
+    title: 'Submission Successful!',
+    confirmButtonText: 'Close',
+    confirmEnabled: true,
+  },
+}
+
+const getSteps = (entityType: EntityType.DOCKER_REPO | EntityType.FILE) => {
+  return entityType === EntityType.DOCKER_REPO ? stepsDocker : stepsFile
+}
+
+type ChallengeSubmissionStepperProps = {
+  projectId: string
+  userId: string
+  teamId: string
+  entity: EntityItem
+  entityType: EntityType.DOCKER_REPO | EntityType.FILE
+  isShowingModal: boolean
+  onClose: () => void
+}
+
+function ChallengeSubmissionStepper({
+  projectId,
+  userId,
+  teamId,
+  entity,
+  entityType,
+  isShowingModal,
+  onClose,
+}: ChallengeSubmissionStepperProps) {
+  const { accessToken } = useSynapseContext()
+  const steps = getSteps(entityType)
+  const initialStep =
+    entityType === EntityType.DOCKER_REPO
+      ? steps.SELECT_COMMIT
+      : steps.SELECT_EVALUATION
+  const [step, setStep] = useState<Step>(initialStep)
+  const [errorMessage, setErrorMessage] = useState<string>()
+  const [selectedCommit, setSelectedCommit] = useState<DockerCommit>()
+
+  const [submissionName, setSubmissionName] = useState<string>('')
+  const [selectedEval, setSelectedEval] = useState<string | undefined>()
+  const [confirming, setConfirming] = useState<boolean>(false)
+
+  const hide = () => {
+    setErrorMessage(undefined)
+    setSelectedCommit(undefined)
+    setSubmissionName('')
+    setSelectedEval(undefined)
+    setConfirming(false)
+    onClose()
+    setStep(initialStep)
+  }
+
+  function handleStepChange(value?: StepsEnum) {
+    if (!value || !steps[value]) return
+    setErrorMessage(undefined)
+    setStep(steps[value])
+  }
+
+  async function confirmEligibility() {
+    if (!selectedEval) return setErrorMessage('Please select an evaluation.')
+    try {
+      return await SynapseClient.getSubmissionEligibility(
+        selectedEval,
+        teamId,
+        accessToken,
+      )
+    } catch (err) {
+      return setErrorMessage(err.message)
+    }
+  }
+
+  async function submitForEvaluation(eligibility: TeamSubmissionEligibility) {
+    if (
+      !entity.id ||
+      (entityType === EntityType.DOCKER_REPO && !selectedCommit)
+    )
+      return setErrorMessage('Error: Invalid entity or commit.')
+
+    const contributors = eligibility.membersEligibility
+      .filter(
+        member =>
+          !member.hasConflictingSubmission &&
+          member.isEligible &&
+          !member.isQuotaFilled &&
+          member.isRegistered,
+      )
+      .map(member => {
+        return { principalId: member.principalId.toString() }
+      })
+
+    const submission: EvaluationSubmission = {
+      userId: userId,
+      evaluationId: selectedEval!,
+      entityId: entity.id,
+      versionNumber: entity.versionNumber ?? 1,
```

Comment:
@jay-hodgson Is this what you're looking for?

Reply:
Yes!  I must have been looking at an old commit.  
Is entity.versionNumber generally defined here?  Still recommend to fallback to an invalid number (like -1) if this must be set, so it will show an error if it's attempting to use the fallback